### PR TITLE
Validation service refactoring

### PR DESCRIPTION
**Pull request by rphmeier**
This PR refactors some of the internal logic of the `validation` crate.

Summary of changes:
  - Validation service is now a future, not a background thread
  - Rather than the block authorship pipeline holding an `Arc<Mutex<>>` to the validation service's state, it now requests info from the validation service by a (bounded) channel.
  - The block authorship pipeline has been extracted into a new file.

### CODE REVIEW

**Comment by rphmeier** [comment]
cc @bkchr can you confirm this works with Cumulus?

File: validation/src/validation_service/mod.rs [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+pub struct ServiceHandle {
+	sender: futures::channel::mpsc::Sender<ValidationInstanceRequest>,
+}
+
+impl ServiceHandle {
+	/// Requests instantiation or cloning of a validation instance from the service.
+	///
+	/// This can fail if the service task has shut down for some reason.
+	pub(crate) fn get_validation_instance(self, relay_parent: Hash)
+		-> impl Future<Output = Result<Arc<ValidationInstanceHandle>, Error>> + Send + 'static
```
Comment (montekki) [inline validation/src/validation_service/mod.rs]: Can this just be an `async` function?

File: validation/src/validation_service/mod.rs [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+		let (tx, rx) = futures::channel::mpsc::channel(CHAN_BUFFER);
+		let interval = interval(TIMER_INTERVAL).map(|_| Message::CollectGarbage);
+		let import_notifications = client.import_notification_stream().map(Message::NotifyImport);
+		let instance_requests = rx.map(Message::RequestInstance);
+		let service = ServiceHandle { sender: tx };
+
+		let background_work = async move {
+			let message_stream = futures::stream::select(interval, instance_requests);
+			let mut message_stream = futures::stream::select(import_notifications, message_stream);
+			loop {
```
Comment (montekki) [inline validation/src/validation_service/mod.rs]: Probably a more idiomatic way:

```rust
while let Some(message) = message_stream.next().await
```

**Comment by rphmeier** [comment]
Thanks Fedor!

**Pull request merged**